lib/kitchen/driver/ec2.rb

```diff
@@ -251,36 +235,60 @@ def destroy(state)
         state.delete(:hostname)
       end
 
-      def lookup_ami(filters_hash)
-        filters = []
-        filters_hash.each do |key, value|
-          filters.push(:name => key.to_s, :values => Array(value))
+      def image
+        return @image if defined?(@image)
+
+        if config[:image_id]
+          @image = ec2.resource.image(config[:image_id])
+          show_chosen_image
+
+        else
+          error("Neither image_id nor an image_search specified for instance #{instance.name}!" \
+                " Please specify one or the other.")
+          raise "Neither image_id nor an image_search specified for instance #{instance.name}!" \
+                " Please specify one or the other."
         end
-        images = ec2.resource.images(:filters => filters).sort do |ami1, ami2|
-          Time.parse(ami1.creation_date) <=> Time.parse(ami2.creation_date)
+
+        @image
+      end
+
+      def default_instance_type
+        @instance_type ||= begin
+          # We default to the free tier (t2.micro for hvm, t1.micro for paravirtual)
+          config[:flavor_id] ||
+            ((image && image.virtualization_type == "hvm") ? "t2.micro" : "t1.micro")
         end
-        images.last && images.last.id
       end
 
-      def ubuntu_ami(region, platform_name)
-        release = amis["ubuntu_releases"][platform_name]
-        Ubuntu.release(release).amis.find do |ami|
-          ami.arch == "amd64" &&
-            ami.root_store == "ebs" &&
-            ami.region == region &&
-            ami.virtualization_type == "hvm"
+      # The actual platform is the platform detected from the image
+      def actual_platform
+        @actual_platform ||= Aws::StandardPlatform.from_image(self, image) if image
+      end
+
+      def desired_platform
+        @desired_platform ||= begin
+          platform = Aws::StandardPlatform.from_platform_string(self, instance.platform.name)
+          if platform
+            debug("platform name #{instance.platform.name} appears to be a standard platform." \
+                  " Searching for #{platform} ...")
+          else
+            debug("platform name #{instance.platform.name} does not have a valid os." \
+                  " Searching for latest stable ubuntu ...")
+            platform = Aws::StandardPlatform.from_platform_string(self, "ubuntu")
```

Comment:
I might be in the minority, but it feels weird to default to ubuntu if we can't figure out what the user wants. Shouldn't we raise an exception here, or at least be very, very vocal about us defaulting to a particular OS?